network/src/lib.rs

```diff
@@ -141,3 +148,105 @@ where
 		Ok(Validation::Success)
 	}
 }
+
+pub struct WaitToAnnounce<Block: BlockT> {
```

Comment:
Some docs would be nice.

Reply:
Yes! I was waiting for you to validate the implementation first to be sure

Reply:
done